File: 3newtonCotes.py
def trapezoidal(file):
    fin = open(file)
    line = fin.readline()
    x0, y0 = [float(v) for v in line.split()]
    I = 0
    for line in fin:
        x, y = [float(v) for v in line.split()]
        I += (x-x0)*(y+y0)/2
        x0 = x
        y0 = y
    return I

I = trapezoidal("data.txt")
print("Integral =", round(I,5))

File: 3newtonCotes.cpp
#include<iostream>
#include<fstream>
using namespace std;

double trapezoidal(string file){
    double x0{}, y0{}, I{};
    ifstream fin{file};
    fin>>x0>>y0;
    for(double x, y; fin>>x>>y;){
        I += (x-x0)*(y+y0)/2;
        x0 = x;
        y0 = y;
    }
    fin.close();
    return I;
}

int main(){
    double I = trapezoidal("data.txt");
    cout<<"Integral = "<<I<<endl;
}
